src/views/studio/studio-comments.jsx

```diff
@@ -30,12 +31,22 @@ const StudioComments = ({
     canRestoreComment,
     handleDeleteComment,
     handleRestoreComment,
+    handleResetComments,
     handleReportComment,
     handleLoadMoreReplies
 }) => {
     useEffect(() => {
         if (comments.length === 0) handleLoadMoreComments();
-    }, []); // Only runs once after the first render
+    }, [comments.length === 0]);
+
+    // The comments you see depend on your admin status
+    // so reset them if isAdmin changes.
+    const adminRef = useRef(isAdmin);
```

Comment:
refs are often used to hold on to DOM nodes, but they actually can contain any mutable value (they return an object with a `.current` property, you can assign whatever you want to it). The first argument to `useRef` is the initial current value, if the ref hasn't already been created. https://reactjs.org/docs/hooks-reference.html#useref